### PR TITLE
feat(worker): an 'external' plugin which supports caching resources that have no build-time hash

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

**Review by gkalpak — COMMENTED** [review]
As far as I understand (which is not far at all :grin:), it LGTM.
It would be nice if the tests ran on Travis (it seems that #120 might help with that).

File: service-worker/worker/src/plugins/external/index.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export function ExternalContentCache(options?: ExternalContentCacheOptions): PluginFactory<ExternalPlugin> {
+  const manifestKey = (options && options.manifestKey) || 'external';
+  return (worker: VersionWorker) => new ExternalPlugin(worker as VersionWorkerImpl, manifestKey);
+}
+
+export class ExternalPlugin implements Plugin<ExternalPlugin> {
+  private cacheKey: string;
+
+  constructor(public worker: VersionWorkerImpl, public key: string) {
```
Comment (gkalpak) [inline service-worker/worker/src/plugins/external/index.ts]: Why `VersionWorkerImpl` instead of `VersionWorker`?

Comment (alxhub) [inline service-worker/worker/src/plugins/external/index.ts]: Originally there was a planned distinction between the public interface of the `VersionWorker` presented to plugins and the `VersionWorkerImpl` implementation actually used. There turned out to be no practical benefit to this, though, so the `VersionWorker` interface is really legacy and needs to be dropped entirely. Everything in practice is done with the concrete type.

File: service-worker/worker/src/test/e2e/spec/sanity.e2e.ts [hunk 2]
```diff
@@ -125,6 +131,13 @@ describe('world sanity', () => {
       .then(result => expect(result).toBe('Hello world!'))
       .then(done), 2000);
   });
+  it('and cached /hello.txt with full url', done => {
```
Comment (alxhub) [inline service-worker/worker/src/test/e2e/spec/sanity.e2e.ts]: Right, will do.

**Pull request closed without merging**